examples_tests/44.LevelCurveExtraction/main.cpp

```diff
@@ -161,16 +160,49 @@ void main() {
 
 }
 )===";
-
-struct SLinesSettings {
-public:
-    float x, y, z;
-    float spacing;
-};
-
 using namespace irr;
 using namespace core;
 
+
+void SaveBufferToCSV(video::IVideoDriver* driver, const uint32_t &bufferBytesize, const uint32_t& alignment,video::IGPUBuffer *bufferToCopy)
+{
+    auto downloadStagingArea = driver->getDefaultDownStreamingBuffer();
+    uint32_t address = std::remove_pointer<decltype(downloadStagingArea)>::type::invalid_address;
+    constexpr uint64_t timeoutInNanoSeconds = 300000000000u;
+    auto unallocatedSize = downloadStagingArea->multi_alloc(std::chrono::nanoseconds(timeoutInNanoSeconds), 1u, &address, &bufferBytesize, &alignment);
+    if (unallocatedSize)
+    {
+        os::Printer::log("Could not download the buffer from the GPU!", ELL_ERROR);
+        return;
+    }
+    driver->copyBuffer(bufferToCopy, downloadStagingArea->getBuffer(), 0u, address, bufferBytesize);
+    auto downloadFence = driver->placeFence(true);
+    auto* data = reinterpret_cast<float*>(downloadStagingArea->getBufferPointer()) + address;
```

Comment:
you forgot how pointer arithmetic works, either `reinterpret_cast<float*>(reinterpret_cast<uint8_t*>(ptr)+address)` or divide `address` by `sizeof(float)`